--- CMakeLists.txt  2023-11-06 19:38:41.444165000 -0300
+++ sdl-gpu/CMakeLists.txt   2023-11-18 14:16:40.825781052 -0300
@@ -74,6 +74,19 @@
 elseif(WIN32)
   message(" Using Windows defaults.")
   set(DEFAULT_INSTALL_LIBRARY OFF)
+elseif(SWITCH)
+  set(DEFAULT_INSTALL_LIBRARY OFF)
+  set(DEFAULT_BUILD_DEMOS OFF)
+  set(DEFAULT_DISABLE_OPENGL ON)
+  set(DEFAULT_DISABLE_GLES OFF)
+  set(DEFAULT_DISABLE_GLES_1 ON)
+  set(DEFAULT_DISABLE_GLES_2 OFF)
+  set(DEFAULT_BUILD_SHARED OFF)
+  set(DEFAULT_BUILD_STATIC ON)
+  include_directories(${INCLUDE_DIRECTORIES})
+  add_definitions(-DGL_GLEXT_PROTOTYPES)
+  add_definitions(-DADD_MISSING_OES_FUNCTIONS)
+  add_definitions(-DSDL_GPU_DISABLE_GLES_1)
 elseif(("${CMAKE_SYSTEM}" MATCHES "Linux") AND ("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "arm"))
   message(" Using Linux ARM defaults.")
 
